### ISSUE TITLE
Bug: it doesn't work when path or filename contains foreign characters

### ISSUE THREAD

**Issue by lemartialou** [prantlf/node-posix-ext#2]
Hi,
First thank you for making this module, you seem to be the only one caring about solving this particular problem ^^

It doesn't work as expected though, I think i found a bug when path or filename contains foreign characters.

```
var posix = require('posix-ext'),
    fs = posix.fs;

    console.log('hello');

fs.stat ('tést-漢字.txt', function(err, stats) {
    if (err) { throw err; }

    console.log(stats);
});
```

The module seems to recognize ascii-us characters only, no accents or special occidental characters allowed, no non-occidental characters allowed (japanese, chinese, arabic...)

The error message that I've got is localized (in french) and is malformed too (accents crippled)

```
if (err) { throw err; }
                 ^
Error: Le fichier sp´┐¢cifi´┐¢ est introuvable.
```

(Instead of "Le fichier spécifié est introuvable", meaning "Specified file not found")

I had some kind of similar problem with another module: 
https://github.com/mscdex/mmmagic/issues/31

And the author came up with this solution:
https://github.com/mscdex/mmmagic/commit/f3e8d14adb458e5d7c134e9c69f86c60b7c5eae8

But I'm not sure it's relevant to this situation though...

**Comment by prantlf** [comment]
Yes, the problem could be, that I send UTF-8 file name to Win32 API.

    DWORD error = GetNamedSecurityInfo((LPSTR) path, SE_FILE_OBJECT,
      OWNER_SECURITY_INFORMATION | GROUP_SECURITY_INFORMATION,
      &usid, &gsid, NULL, NULL, &sd);
    if (error != ERROR_SUCCESS) {
      return error;
    }

The error message is not so clear to me, because I use `WinapiErrnoException` from Node.js. I will try to debug it.

    if (async_data->error != ERROR_SUCCESS) {
      argv[0] = WinapiErrnoException(async_data->error);
    } else {

I hope, that I will be able to reproduce and fix it in Czech environment.

Thanks a lot for the hit in your referred issues!

**Comment by prantlf** [comment]
I've released 0.3.2 with the fix for passing the input parameters in UTF-8. I realised, that I converted output parameters, but not the input ones. Thanks for the help!

Error messages may contain some malformed characters. I installed Czech language pack and noticed, that while Czech characters were printed correctly, Chinese characters were still malformed. It appears, that it depends on capabilities of the used console/terminal.
